install_gridpack_deps.sh

```diff
@@ -1,133 +1,168 @@
-then
+#! /bin/bash
 
-  rm -rf boost*
-    
-  # Download and install Boost
-  echo "Downloading Boost-1.78.0"
+# installs GridPACK dependencies to the current directory
+
+set -xeuo pipefail
+
+function install_boost {
+  local boost_version=$1
 
-  # Download Boost
-  wget https://boostorg.jfrog.io/artifactory/main/release/1.78.0/source/boost_1_78_0.tar.gz
+  # check args
+  : "${boost_version:?}"
 
-  # Untar
-  tar -xf boost_1_78_0.tar.gz
+  echo "--- Installing Boost ${boost_version} ---"
+
+  # remove existing
+  rm -rf boost*
 
-  cd boost_1_78_0
+  # download
+  echo "Downloading Boost"
+  wget \
+    "https://boostorg.jfrog.io/artifactory/main/release/${boost_version}/source/boost_${boost_version//./_}.tar.gz" \
+    -O boost.tar.gz \
+    --quiet
 
-  # Install boost
-  echo "Building Boost-1.78.0"
+  # unpack
+  echo "Unpacking Boost"
+  tar -xf boost.tar.gz && rm boost.tar.gz
 
-  ./bootstrap.sh --prefix=install_for_gridpack --with-libraries=mpi,serialization,random,filesystem,system
+  # remove version from dir name
+  mv "boost_${boost_version//./_}" boost
 
-  echo 'using mpi ;' >> project-config.jam
+  pushd boost || exit
 
+  # bootstrap
+  echo "Bootstrapping Boost"
+  ./bootstrap.sh \
+    --prefix=install_for_gridpack \
+    --with-libraries=mpi,serialization,random,filesystem,system
+  echo 'using mpi ;' >>project-config.jam
+
+  # build
+  echo "Building Boost"
   ./b2 -a -d+2 link=shared stage
 
-  echo "Installing Boost-1.78.0"
+  # install
+  echo "Installing Boost"
   ./b2 -a -d+2 link=shared install
 
-  echo "Building and Installing Boost libraries complete"
-fi
+  popd || exit
+
+  echo "Boost installation complete"
+}
+
+function install_ga {
+  local ga_version=$1
+
+  # check args
+  : "${ga_version:?}"
 
-if ${install_ga}
-then
-  # Download, build, and install GA
-  cd ${GP_EXT_DEPS}
+  echo "--- Installing Global Arrays ${ga_version} ---"
 
-  echo "Downloading GA-5.8"
+  # download
+  echo "Downloading Global Arrays"
+  wget \
+    "https://github.com/GlobalArrays/ga/releases/download/v${ga_version}/ga-${ga_version}.tar.gz" \
+    -O ga.tar.gz \
+    --quiet
 
-  wget https://github.com/GlobalArrays/ga/releases/download/v5.8/ga-5.8.tar.gz
+  # unpack
+  echo "Unpacking Global Arrays"
+  tar -xf ga.tar.gz && rm ga.tar.gz
 
-  tar -xf ga-5.8.tar.gz
+  # remove version from dir name
+  mv "ga-${ga_version}" ga
 
-  cd ga-5.8
+  pushd ga || exit
 
-  # Build GA
-  echo "Building GA-5.8"
-  ./configure --with-mpi-ts --disable-f77 --without-blas --enable-cxx --enable-i4 --prefix=${PWD}/install_for_gridpack --enable-shared
-  
-  # Install GA
-  echo "Installing GA-5.8"
-  make -j 10 install
-  
-  echo "GA-5.8 installation complete"
-fi
+  # check if we are running on RHEL
+  distribution=$(
+    # shellcheck source=/dev/null
+    source /etc/os-release
+    echo "$ID"
+  )
+  case $distribution in
+  fedora | rhel | centos | rocky)
+    echo "Loading mpi4py module"
+    module load "mpi/openmpi-$(arch)"
+    ;;
```

Comment:
As above, the echoed message is misleading.  `module` doesn't affect mpi4py.

Reply:
Fixed